connectors/sources/confluence.py

```diff
@@ -204,6 +235,39 @@ async def _handle_client_errors(self, url, exception):
         else:
             raise
 
+    async def _handle_client_payload_error(self, exception):
+        retry_seconds = DEFAULT_RETRY_SECONDS
+        response_headers = exception.headers or {}
+        if "Retry-After" in response_headers:
```

Comment:
What about client payload errors makes it special and the only thing that should look at a retry-after header?

Reply:
Client payload errors often include a Retry-After header to manage rate limits and server load, indicating how long to wait before retrying, similar to how we handle it for OneDrive and SharePoint Online.

Reply:
But why not other error types too? Like 429 response codes should pretty much always include a `Retry-After` header

Reply:
You're right, 429 response codes should also include a Retry-After header. However, we already handle this logic in the `_handle_client_errors()`, which checks for the Retry-After header for 429 response code